test/nuts/deployDestructive.nut.ts

```diff
@@ -0,0 +1,140 @@
+      nut: __filename,
+      executable: os.platform() === 'win32' ? executable.replace(/\\/g, '\\\\') : executable,
+      repository: 'https://github.com/trailheadapps/dreamhouse-lwc.git',
+    });
+    execCmd('force:source:deploy --sourcepath force-app');
+  });
+
+  after(async () => {
+    await testkit?.clean();
+  });
+
+  describe('destructive changes POST', () => {
+    it('should deploy and then delete an ApexClass ', async () => {
+      const { apexName } = createApexClass();
+      let soql = query('ApexClass', apexName);
+
+      expect(soql.result.records[0].IsNameObsolete).to.be.false;
+      createManifest('ApexClass:GeocodingService', 'package');
+      createManifest(`ApexClass:${apexName}`, 'post');
+
+      execCmd('force:source:deploy --json --manifest package.xml --postdestructivechanges destructiveChangesPost.xml', {
+        ensureExitCode: 0,
+      });
+
+      soql = query('ApexClass', apexName);
+      expect(soql.result.records[0].IsNameObsolete).to.be.true;
+    });
+  });
+
+  describe('destructive changes PRE', () => {
+    it('should delete an ApexClass and then deploy a class', async () => {
+      const { apexName } = createApexClass();
+      let soql = query('ApexClass', apexName);
+
+      expect(soql.result.records[0].IsNameObsolete).to.be.false;
+      createManifest('ApexClass:GeocodingService', 'package');
+      createManifest(`ApexClass:${apexName}`, 'pre');
+
+      execCmd('force:source:deploy --json --manifest package.xml --predestructivechanges destructiveChangesPre.xml', {
+        ensureExitCode: 0,
+      });
+
+      soql = query('ApexClass', apexName);
+      expect(soql.result.records[0].IsNameObsolete).to.be.true;
+    });
+  });
+
+  describe('destructive changes POST and PRE', () => {
+    it('should delete a class, then deploy and then delete an ApexClass', async () => {
+      const pre = createApexClass('pre').apexName;
+      const post = createApexClass('post').apexName;
+      let soqlPre = query('ApexClass', pre);
+      let soqlPost = query('ApexClass', post);
+
+      expect(soqlPre.result.records[0].IsNameObsolete).to.be.false;
+      expect(soqlPost.result.records[0].IsNameObsolete).to.be.false;
+      createManifest('ApexClass:GeocodingService', 'package');
+      createManifest(`ApexClass:${post}`, 'post');
+      createManifest(`ApexClass:${pre}`, 'pre');
+
+      execCmd(
+        'force:source:deploy --json --manifest package.xml --postdestructivechanges destructiveChangesPost.xml --predestructivechanges destructiveChangesPre.xml',
+        {
+          ensureExitCode: 0,
+        }
+      );
+
+      soqlPre = query('ApexClass', pre);
+      soqlPost = query('ApexClass', post);
+      expect(soqlPre.result.records[0].IsNameObsolete).to.be.true;
+      expect(soqlPost.result.records[0].IsNameObsolete).to.be.true;
+    });
+  });
+
+  describe('errors', () => {
+    it('should throw an error when a destructive flag is passed without the manifest flag', () => {
```

Comment:
what happens when a value of destructive flag is passed, but the file doesn't exist (typo, or user in wrong directory level, etc)